packages/app-project/src/screens/ClassifyPage/components/ClassifierWrapper/ClassifierWrapper.js

```diff
@@ -81,22 +77,21 @@ export default function ClassifierWrapper({
   if (subjectSetID) {
     baseURL = `${baseURL}/subject-set/${subjectSetID}`
   }
-
-  if (router.query.subjectID && classifierSubjectID) {
-    const subjectPageURL = `${baseURL}/subject/${classifierSubjectID}`
-    /* True when the classifier subject changed, but new id is not yet reflected in the url */
-    if (router?.query.subjectID !== classifierSubjectID) {
-      const href = addQueryParams(subjectPageURL)
-      router.push(href, href, { shallow: true })
-    }
-  }
+  const subjectFromURL = router?.query.subjectID
 
   /*
     Track the current classification subject, when it changes inside the classifier.
   */
   const onSubjectChange = useCallback((subject) => {
-    setClassifierSubjectID(subject?.id)
-  }, [])
+    if (subjectFromURL) {
+      const subjectPageURL = `${baseURL}/subject/${subject.id}`
+      /* True when the classifier subject changed, but new id is not yet reflected in the url */
+      if (subjectFromURL !== subject.id) {
+        const href = addQueryParams(subjectPageURL)
+        router.replace(href, href, { shallow: true })
```

Comment:
This could also be `router.push`. Depends how we want the browser Back button to behave.